### PR TITLE
Dronekit-python mission check script

### PR DESCRIPTION
**Pull request by sanderux**
First shot at simple mission checking.
- loads a mission
- waits until the mission sequence has reached the amount of mission items loaded
- waits until the vehicle has landed
- monitors max execution time
- monitors waypoint altitude adherence

It requires a recent version of dronekit 

```
usage: MissionCheck.py [-h] [-c CONNECT] [-f FILENAME] [-t TIMEOUT]
                       [-a ALTRAD]

optional arguments:
  -h, --help            show this help message and exit
  -c CONNECT, --connect CONNECT
                        connection string
  -f FILENAME, --filename FILENAME
                        mission filename
  -t TIMEOUT, --timeout TIMEOUT
                        execution timeout
  -a ALTRAD, --altrad ALTRAD
                        altitude acceptance radius
```

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@AndreasAntener Could you help to get that on Jenkins?

File: Tools/CI/MissionCheck.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+from dronekit import connect, VehicleMode, Command
+import time, sys
+
+# start time counter
+start_time = time.time()
+elapsed_time = time.time() - start_time
+
+# Connect to the Vehicle, wait_ready will not work on PX4
+print "Connecting"
+vehicle = connect('127.0.0.1:14550', wait_ready=True)
```
Comment (LorenzMeier) [inline Tools/CI/MissionCheck.py]: Does it still work if you use 14540 here? That would connect to the off board control port and leave the other one open to check the progress with QGroundControl.

**Comment by sanderux** [comment]
Yes, it does.. i' adding a few changes and consts. I'll make the port configurable

**Comment by LorenzMeier** [comment]
@peterbarker @hamishwillee @mrpollo This is the current WIP to send a PX4 system onto a mission. Might be educational for you to see what shim you need to make DroneKit fully MAVLink compatible (either that or you'll find all the PX4 quirks we need to fix) and not implicitly assuming talking to APM.

**Comment by sanderux** [comment]
I think that fixing the flight modes in pymavlink/mavutil.py will go a very long way

File: Tools/CI/MissionCheck.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+vehicle.armed = False
+
+while not vehicle.system_status.state == "STANDBY":
```
Comment (mrpollo) [inline Tools/CI/MissionCheck.py]: we should probably create a callback to handle vehicle state correctly

**Comment by mrpollo** [comment]
hey @LorenzMeier this is great, I would suggest we use a test framework for this type of scripts, on dronekit-python we are using dronekit-sitl to create integration tests, we should probably look into adding PX4 support to that as well.

PX4 support on DroneKit is long overdue, let me know how can we help meet your goals now, and we should talk about how to proceed going forward.

**Comment by hamishwillee** [comment]
This looks a lot like our mission import/export example: http://python.dronekit.io/examples/mission_import_export.html

Knowing whether that works for talking to PX4 and whether your script works on ArduPilot will be interesting.

**Comment by peterbarker** [comment]
peter.barker
11:30 AM Interesting in that PX4 dronekit-python example they change mode to AUTO ​then​ arm.  You can't do that in ArduPilot.  I wonder if you can arm in STANDBY on PX4.... if not, that's going to be an interesting one to solve for dronekit-python.

**Comment by LorenzMeier** [comment]
@sanderux Could you also send a PR for DK installation instructions for http://dev.px4.io and add a shell script here which runs the simulator and script in the right order, so it all gets automated?

**Comment by AndreasAntener** [comment]
@LorenzMeier @sanderux we might need to look at the sequence switching to auto then arming. I'm afraid in reality this still leads to issues with takeoff as the arming state changes before the props spin.

**Comment by sanderux** [comment]
I'll change the order and look at a full script and docs

**Comment by sanderux** [comment]
@Hamishwillee the mission read/write functions have been taken from the examples indeed. Should that be credited?

**Comment by hamishwillee** [comment]
@sanderux Probably - this is Apache 2 (free use) but you should retain the Copyright information (as I understand the license). We're very happy for them to be reused by you. My main interest is easily being able to spot the differences that you needed to make so we can make DroneKit better.

**Comment by sanderux** [comment]
@hamishwillee I'm sorry, I did not find any copyright information quoted in the example script (https://github.com/dronekit/dronekit-python/blob/master/examples/mission_import_export/mission_import_export.py) What additional header would you like me to add to this?

**Comment by hamishwillee** [comment]
@sanderux Note your fault - there isn't any on the examples, but there is on the documentation itself. I'll add a copyright to the files too at some point (https://github.com/dronekit/dronekit-python/issues/568)

in the meantime, please add to your files:
© Copyright 2015-2016, 3D Robotics.

**Comment by sanderux** [comment]
@AndreasAntener or @LorenzMeier I tried to add a launch script that starts SITL, executes missioncheck and ends SITL. I did not succeed on the last part.

Any help appreciated

**Comment by LorenzMeier** [comment]
Rebased and applied.

**Pull request closed without merging**